File: shoring/compat/src/test/java/com/panopset/compat/test/FlagSwitchTest.java
package com.panopset.compat.test;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import com.panopset.compat.FlagSwitch;

public class FlagSwitchTest {

  @Test
  void test() {
    FlagSwitch flag = new FlagSwitch();
    Assertions.assertTrue(flag.pull());
    Assertions.assertFalse(flag.pull());
    Assertions.assertTrue(flag.pull());
    flag.reset();
    Assertions.assertTrue(flag.pull());
  }
}

File: code/shoring/compat/src/test/kotlin/com/panopset/compat/test/FlagSwitchTest.kt
package com.panopset.compat.test

import com.panopset.compat.FlagSwitch
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class FlagSwitchTest {
    @Test
    fun test() {
        val flag = FlagSwitch()
        Assertions.assertTrue(flag.pull())
        Assertions.assertFalse(flag.pull())
        Assertions.assertTrue(flag.pull())
        flag.reset()
        Assertions.assertTrue(flag.pull())
    }
}